### PR TITLE
Adds obsv samples for CSI Powermax

### PR DESCRIPTION
**Pull request by delldubey**
# Description
- Add Observability for CSI Powermax
- Add unit tests for powermax related changes in observability.
- 
# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/908|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
- [x] Unit Tests
- [x] Cluster Tests

### CODE REVIEW

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]
Need some e2e tests for PowerMax + Observability -- write similar tests to what we have for PowerFlex + Obs and PowerScale + Obs
Add cert-manager to yamls as per this PR: https://github.com/dell/csm-operator/pull/310

If you need any help or want to talk through any of this, hmu on slack

**Review by shanmydell — APPROVED** [review]

**Comment by delldubey** [comment]
> Need some e2e tests for PowerMax + Observability -- write similar tests to what we have for PowerFlex + Obs and PowerScale + Obs Add cert-manager to yamls as per this PR: #310
> 
> If you need any help or want to talk through any of this, hmu on slack

Added.

**Review by shanmydell — APPROVED** [review]

**Review by boyamurthy — APPROVED** [review]
LGTM.

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]

**Comment by jooseppi-luna** [comment]
Other than my one comment everything looks good, if you can address it by 6:00 AM EST/3:30 PM IST I can approve

**Review by boyamurthy — DISMISSED** [review]
LGTM.

**Review by boyamurthy — APPROVED** [review]
LGTM.

**Review by jooseppi-luna — APPROVED** [review]

**Comment by santhoshatdell** [comment]
It looks _operatorconfig/moduleconfig/common/version-values.yaml_ is not updated for PowerMax + Obs.
Shouldn't we also need karavi-metrics-powermax.yaml file for Obs 1.6.0 ?

+CC  @donatwork 

image

File: samples/storage_csm_powermax_v280.yaml [hunk 1]
```diff
@@ -327,3 +327,93 @@ spec:
         # image: Defines replication init container image. This shouldn't be changed
         # Allowed values: string
         image: dellemc/dell-replication-init:v1.0.1
+
+    # observability: allows to configure observability
+    - name: observability
+      # enabled: Enable/Disable observability
+      enabled: false
+      configVersion: v1.5.0
```
Comment (donatwork) [inline samples/storage_csm_powermax_v280.yaml:335]: Should be v1.6.0.

File: samples/storage_csm_powermax_v280.yaml [hunk 2]
```diff
@@ -327,3 +327,93 @@ spec:
...
+          #   true: enable deployment of cert-manager
+          #   false: disable deployment of cert-manager only if it's already deployed
+          # Default value: false
+          enabled: false
+
+        - name: metrics-powermax
+          # enabled: Enable/Disable PowerMax metrics
+          enabled: false
+          # image: Defines PowerMax metrics image. This shouldn't be changed
+          image: dellemc/csm-metrics-powermax:v1.1.0
```
Comment (donatwork) [inline samples/storage_csm_powermax_v280.yaml:375]: Should be v1.2.0.

File: samples/storage_csm_powermax_v280.yaml [hunk 3]
```diff
@@ -327,3 +327,93 @@ spec:
...
+      # enabled: Enable/Disable observability
+      enabled: false
+      configVersion: v1.5.0
+      components:
+        - name: topology
+          # enabled: Enable/Disable topology
+          enabled: false
+          # image: Defines karavi-topology image. This shouldn't be changed
+          # Allowed values: string
+          image: dellemc/csm-topology:v1.5.0
```
Comment (donatwork) [inline samples/storage_csm_powermax_v280.yaml:342]: Shoudl be v1.6.0.

**Review by donatwork — COMMENTED** [review]
The versions need updating.

**Pull request merged**